WooCommerce/Classes/ViewRelated/Blaze/ConfirmPayment/BlazeConfirmPaymentViewModel.swift

```diff
@@ -21,18 +21,27 @@ final class BlazeConfirmPaymentViewModel: ObservableObject {
         isFetchingPaymentInfo || selectedPaymentMethod == nil
     }
 
+    @Published var showAddPaymentSheet: Bool = false
+
     var paymentMethodsViewModel: BlazePaymentMethodsViewModel? {
         guard let paymentInfo else {
             DDLogError("⛔️ No payment info available to list in payment methods screen.")
             return nil
         }
         return BlazePaymentMethodsViewModel(siteID: siteID,
                                             paymentInfo: paymentInfo,
-                                            selectedPaymentMethodID: selectedPaymentMethod?.id, completion: { paymentID in
+                                            selectedPaymentMethodID: selectedPaymentMethod?.id,
+                                            completion: { paymentID in
             Task { @MainActor [weak self] in
                 guard let self else { return }
-                await updatePaymentInfo()
-                selectedPaymentMethod = paymentInfo.savedPaymentMethods.first(where: { $0.id == paymentID })
+                showAddPaymentSheet = false
+
+                if let paymentIDExisting = paymentInfo.savedPaymentMethods.first(where: { $0.id == paymentID }) {
```

Comment:
Nit:

```suggestion
                if let existingPaymentMethod = paymentInfo.savedPaymentMethods.first(where: { $0.id == paymentID }) {
```

Reply:
Done in https://github.com/woocommerce/woocommerce-ios/pull/11754/commits/c013202b91dacc653e4803b8a1e0904ec5972df6